File: src/main/java/com/badbones69/crazycrates/config/Locale.java
package com.badbones69.crazycrates.config;

import com.badbones69.crazycrates.api.files.AbstractConfig;
import com.badbones69.crazycrates.utils.FileUtils;
import net.kyori.adventure.audience.Audience;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.minimessage.MiniMessage;
import net.kyori.adventure.text.minimessage.tag.resolver.TagResolver;
import org.bukkit.command.ConsoleCommandSender;

public class Locale extends AbstractConfig {

    private static final Locale LOCALE = new Locale();

    public static void reload() {
        LOCALE.reload(FileUtils.INSTANCE.getLocaleDir().resolve(Config.languageFile), Locale.class);
    }

    public static void send(Audience recipient, String msg, TagResolver.Single... placeholders) {
        send(recipient, true, msg, placeholders);
    }

    public static void send(Audience recipient, boolean prefix, String msg, TagResolver.Single... placeholders) {
        if (msg == null) return;

        for (String part : msg.split("\n")) {
            send(recipient, prefix, parse(part, placeholders));
        }
    }

    public static void send(Audience recipient, boolean prefix, Component component) {
        if (recipient instanceof ConsoleCommandSender) {
            recipient.sendMessage(prefix ? parse("bleh").append(component) : component);
            return;
        }

        recipient.sendMessage(prefix ? parse("blah").append(component) : component);
    }

    public static Component parse(String msg, TagResolver.Single... placeholders) {
        return MiniMessage.miniMessage().deserialize(msg, placeholders);
    }
}

File: common/src/main/kotlin/com/badbones69/crazycrates/config/Locale.kt
package com.badbones69.crazycrates.config

import com.badbones69.crazycrates.files.AbstractConfig
import net.kyori.adventure.audience.Audience
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.minimessage.MiniMessage
import net.kyori.adventure.text.minimessage.tag.resolver.TagResolver
import java.nio.file.Path

object Locale : AbstractConfig() {

    fun reload(dataFolder: Path) = reload(dataFolder.resolve(Config().languageFile), this::class.java)

    fun send(recipient: Audience, msg: String, vararg placeholders: TagResolver.Single) = send(recipient, true, msg, *placeholders)

    fun send(recipient: Audience, prefix: Boolean, msg: String, vararg placeholders: TagResolver.Single) {
        msg.split("\n").forEach {
            send(recipient, prefix, parse(it, *placeholders))
        }
    }

    fun send(recipient: Audience, component: Component?) = send(recipient, true, component)

    fun send(recipient: Audience, prefix: Boolean, component: Component?) = recipient.sendMessage((if (prefix) parse("").append(component!!) else component)!!)

    private fun parse(msg: String?, vararg placeholders: TagResolver.Single): Component {
       return MiniMessage.miniMessage().deserialize(msg.toString(), *placeholders)
    }
}